Lib/test/test_functools.py

```diff
@@ -403,6 +403,33 @@ def test_attributes_unwritable(self):
         else:
             self.fail('partial object allowed __dict__ to be deleted')
 
+    def test_manually_adding_non_string_keyword(self):
+        p = self.partial(capture)
+        # Adding a non-string/unicode keyword to partial kwargs
+        p.keywords[1] = 10
+        self.assertIn('1=10', repr(p))
+        try:
```

Comment:
That failed because `assertRaises` needs the `__name__` of the class/function. But `_functools.partial` doesn't have a `__name__`.